snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeResources.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.snowflake;
+
+final class SnowflakeResources {
```

Comment:
Yeah, this is a bit awkward at the moment just because we ideally wanted to create a clean `SnowflakeClient` abstraction layer that can be well-encapsulated from `SnowflakeCatalog` and Iceberg-specific business logic, but indeed the processing of Namespace levels unfortunately leaks into both.

I'll see if refactoring out a SnowflakeIdentifier can move all Namespace-processing stuff into one place and hopefully also make this container class unnecessary.